### ISSUE TITLE
Echo questions

### ISSUE THREAD

**Issue by nschneid** [UniversalDependencies/docs#390]
Mark Steedman writes:

> There is a class of "echo questions" discussed by Huddleston and Pullum, as in B's responses below:
> 
> A: I bought a Picasso!
> 
> B: You bought a PICASSO?
> B': You bought a WHAT?
> B'': You bought WHAT?
> B''': You WHAT?
> B'''': WHAT?
> 
> They are grammatically (and intonationally) distinct from real questions, like "did you buy a Picasso?" and "What did you buy?" and "What did you do?".

What relations should we use for "WHAT" in the above examples? Should we treat B' like a normal noun, B'' like a normal NP, B''' like a normal VP, and B'''' like a normal sentence—or is there a more elegant solution?

**Comment by jnivre** [comment]
Syntactically, WHAT is "obj" in both B' and B'' and "root" in B''' and B''''. Morphologically, it is PRON as usual in all cases.

**Comment by nschneid** [comment]
Mark again:

> I don't see how "what" can be SYNTACTICALLY  object in B': You bought a WHAT?
>
> You can't extract it.  You can't passivize it.
> 
> *A what did you buy?
> *What did you buy a?
> *Picasso was bought a (by Mary)

**Comment by amir-zeldes** [comment]
@nschneid I think the phrase is an object either way; those arguments about extraction and passivization without the article apply to common nouns too:

*Dog did you buy the?
The dog, did you buy? (probably only possible using Yiddish movement)
*Dog was bought a (by Mary)
A dog was bought (by Mary)

The object is the entire phrase [a what], which is headed by what (and the head has the function 'obj' as pointed out by @jnivre )

I think you can move the entire phrase, just like most NPs, and transform using a cleft etc.: 

It was [a WHAT] that you bought?
The thing you bought was [a WHAT]?

The only open question I see is whether it's still to be tagged as PRON. In PTB it would still be WP, so I would be for PRON too. It really depends on your research question or application when searching for stuff like this, but my expectation is that when I look for PRON, I get all pronouns, which would include this case (it is a pronoun in so far as its denotation is completely contextual, because it stands in for a nonu). I could imagine a context in which I don't want it (e.g. I'm looking for all nominal phrase contexts that can/cannot license an article), but I think a robotic what -> PRON decision is more predictable and easier to enforce and validate.

**Comment by amir-zeldes** [comment]
Oh and for the non-NP cases I'd do:

What? root(what)
You what? root(what), nsubj(what,you)
You bought what? obj(bought,what) 
You bought a what? obj(bought,what), det(what,a)

**Comment by nschneid** [comment]
> those arguments about extraction and passivization without the article apply to common nouns too

Yes, I think the more relevant observation is that it's odd to passivize the entire NP:

- non-echo:
  A Picasso was bought (by Mary)
  The Picasso was bought (by Mary)
- echo:
  ?A WHAT was bought (by Mary)
  ?The WHAT was bought (by Mary)

"A WHAT was bought?!" certainly sounds more marginal than "She bought a WHAT?!" But I don't know if the marginality reflects the syntactic relation or the semantic/discourse properties of echo questions.

In other words: I think echo questions are a distinct construction (with special semantics/pragmatics and intonation), but I don't know that it needs a special syntactic relation in the UD context.

**Comment by nschneid** [comment]
I cannot think of other constructions where a pronoun is preceeded by an article: *the it, *a something, *the who ([rock band names](https://en.wikipedia.org/wiki/The_Who) excepted). This gives me pause about `det(what, a)`. According to [this query](http://bionlp-www.utu.fi/dep_search/query?search=who%7Cwhom%7Cwhat%20%3Edet%20_&db=UD_English-dev&case_sensitive=True&hits_per_page=10), there are no instances of *who*, *whom*, or *what* taking a determiner in the English UD treebank. The relationship between "a" and "what" almost feels like [parataxis](http://universaldependencies.org/u/dep/parataxis.html) to me.

**Comment by nschneid** [comment]
OK, the pronoun "one" can follow an article: definite determiners as in [this query](http://bionlp-www.utu.fi/dep_search/query?search=one%20%3Edet%20_&db=UD_English-dev&case_sensitive=True&hits_per_page=10), and the expression ["not a one"](https://en.wiktionary.org/wiki/not_a_one). But it is not a WH-pronoun, and it is actually tagged as a noun in the treebank.

**Comment by nschneid** [comment]
Aha—Huddleston & Pullum, p. 891: "In _You need to buy a new what?_, the echo-question word _what_ is a common noun with a determiner and adjectival modifier as dependents: interrogative _what_, by contrast, is a pronoun, unable to take such dependents." I'd feel better about `det(what, a)` if "what" was tagged as a noun.

**Comment by dan-zeman** [comment]
In UD v2 we do not change the UPOSTAG each time the function of a word changes. So _what_ can stay `PRON` in UD even if Huddleston & Pullum call it a common noun in a specific construction.

**Comment by jnivre** [comment]
I did not mean that "WHAT" by itself is the object in "You bought a WHAT". I only meant that the incoming dependency is labeled "obj". Thus:

nsubj(bought, You)
obj(bought, WHAT)
det(what, a)

But I think Amir made this clear in his comments.

**Comment by manning** [comment]
@nschneid They're marginal in various ways, but there are other constructions with a determiner before a pronoun.  E.g.: "I was hoping for a she in that position." "A ship is called a she because ..." I think the dubiousness/lack of passivization has a reasonable discourse explanation: This is a focusing construction whereas passivization topicalizes the subject. 

I think there is an agreed analysis here, so I'll close this....
